### PR TITLE
Replace QT's JSON library with nlohmann's JSON library

### PR DESCRIPTION
**Pull request by Sightem**
This pull request replaces the usage of QT's JSON library with nlohmann's JSON library in order to improve code readability and maintainability. QT's JSON library syntax can be unintuitive and make code difficult to understand, whereas nlohmann's JSON library provides a more straightforward and easy-to-use syntax

### CODE REVIEW

File: launcher/meta/BaseEntity.cpp [hunk 1]
```diff
@@ -97,9 +104,10 @@ bool Meta::BaseEntity::loadLocalFile()
...
-        auto doc = Json::requireDocument(fname, fname);
-        auto obj = Json::requireObject(doc, fname);
-        parse(obj);
+        //auto doc = Json::requireDocument(fname, fname);
+        //auto obj = Json::requireObject(doc, fname);
+        const auto json = nlohmann::json::parse(std::ifstream(fname.toStdString()));
+        parse(json);
         return true;
     }
     catch (const Exception &e)
```
Comment (ghost) [inline launcher/meta/BaseEntity.cpp]: I believe you need to catch nlohmann::json::parse_error here

File: launcher/meta/BaseEntity.cpp [hunk 2]
```diff
@@ -52,10 +53,16 @@ class ParsingValidator : public Net::Validator
         auto fname = m_entity->localFilename();
         try
```
Comment (ghost) [inline launcher/meta/BaseEntity.cpp]: Outer try statement is redundant.

File: launcher/meta/BaseEntity.cpp (hunk 1 above)
Comment (Sightem) [reply launcher/meta/BaseEntity.cpp]: yes, there are probably some more that I've missed/have used std::exception lazily, will get to them

File: launcher/meta/BaseEntity.cpp (hunk 2 above)
Comment (Sightem) [reply launcher/meta/BaseEntity.cpp]: Addressed

**Comment by slendidev** [comment]
You should not keep unused code commented. Makes the code harder to read and it's no longer needed. Just delete it.

**Comment by Sightem** [comment]
> You should not keep unused code commented. Makes the code harder to read and it's no longer needed. Just delete it.

They're temporary, nvim's git integration plugin sucks and sometimes breaks when I wanna view the changes, they will be removed once I finish thoroughly testing the changes

**Comment by ghost** [comment]
> QT's JSON library syntax can be unintuitive and make code difficult to understand

Seems quite intuitive to me. You just insert stuff.

**Comment by LennyMcLennington** [comment]
Cool idea but the only way this would ever work is if we fork nlohmann json to have the same level of forgiveness as Qt json (since Fabric's official mod template produces invalid JSON and they won't merge a 4 line change to fix it when I pr'd it) at which point why not just use Qt json

**Pull request closed without merging**